blocks/api/factory.js

```diff
@@ -67,6 +68,48 @@ export function createBlock( name, blockAttributes = {} ) {
 }
 
 /**
+ * Returns an array of possible block transformations that could happen on the set of blocks received as argument.
+ *
+ * @param  {Array}  blocks Blocks array
+ * @return {Array}         Array of possible block transformations
+ */
+export function getPossibleBlockTransformations( blocks ) {
+	if ( ! blocks || ! blocks[ 0 ] ) {
+		return [];
+	}
+	const isMultiBlock = blocks.length > 1;
+	const sourceBlockName = blocks[ 0 ].name;
+
+	if ( isMultiBlock && ! every( blocks, ( block ) => ( block.name === sourceBlockName ) ) ) {
+		return [];
+	}
+
+	const blockType = getBlockType( sourceBlockName );
+	const blocksToBeTransformedFrom = reduce( getBlockTypes(), ( memo, type ) => {
+		const transformFrom = get( type, 'transforms.from', [] );
+		const transformation = find(
+			transformFrom,
+			t => t.type === 'block' && t.blocks.indexOf( sourceBlockName ) !== -1 &&
+				( ! isMultiBlock || t.isMultiBlock )
+		);
+		return transformation ? memo.concat( [ type.name ] ) : memo;
+	}, [] );
+	const blocksToBeTransformedTo = get( blockType, 'transforms.to', [] )
+		.reduce(
+			( memo, transformation ) =>
+				memo.concat( ! isMultiBlock || transformation.isMultiBlock ? transformation.blocks : [] ),
```

Comment:
Can't we just assign `blocksToBeTransformedTo` to `blockType.transforms.to` if `isMultiBlock` and skip the `Array#reduce` for those cases where it's constant?